File: src/orion_spur/common/material/PlayerDto.java
package orion_spur.common.material;

import java.util.List;

public class PlayerDto 
{
	private String Name;
	private CoordinateDto X;
	private CoordinateDto Y;
	private float Rotation;

	public PlayerDto(){}

	public PlayerDto(String Name, CoordinateDto X, CoordinateDto Y, float Rotation)
	{
		this.Name = Name;
		this.X = X;
		this.Y = Y;
		this.Rotation = Rotation;
	}

	public String getName()
	{
		return Name;
	}

	public CoordinateDto getX()
	{
		return X;
	}

	public CoordinateDto getY()
	{
		return Y;
	}

	public float getRotation()
	{
		return Rotation;
	}
}

File: server/src/material/PlayerDto.go
package material

type PlayerDto struct {
	Name  string `json:"Name"`
	X  CoordinateDto `json:"X"`
	Y  CoordinateDto `json:"Y"`
	Rotation  float32 `json:"Rotation"`
}

func NewPlayerDto(name string, x CoordinateDto, y CoordinateDto, rotation float32)PlayerDto{
	return PlayerDto{Name: name, X: x, Y: y, Rotation: rotation}
}

func (n PlayerDto) GetName() string {
	return n.Name
}

func (x PlayerDto) GetX() CoordinateDto {
	return x.X
}

func (y PlayerDto) GetY() CoordinateDto {
	return y.Y
}

func (r PlayerDto) GetRotation() float32 {
	return r.Rotation
}


